bin/import_slr88.py

```diff
@@ -0,0 +1,302 @@
+ARCHIVE_EXT = ".tgz"
+ARCHIVE_NAME_txt = "txt"
+ARCHIVE_NAME_wav = "wav"
+BASE_SLR_URL = "http://www.openslr.org/resources/88"
+ARCHIVE_URL_txt = f"{BASE_SLR_URL}/{ARCHIVE_NAME_txt}{ARCHIVE_EXT}"
+ARCHIVE_URL_wav = f"{BASE_SLR_URL}/{ARCHIVE_NAME_wav}{ARCHIVE_EXT}"
+ARCHIVE_ROOT_PATH = "Volumes/CLEM_HDD/IRCAM/Open_SLR"
+
+
+def _download_and_preprocess_data(target_dir):
+    # Making path absolute
+    target_dir = os.path.abspath(target_dir)
+    # Conditionally download data
+    txt_archive_path = maybe_download(f"{ARCHIVE_NAME_txt}{ARCHIVE_EXT}", target_dir, ARCHIVE_URL_txt)
+    wav_archive_path = maybe_download(f"{ARCHIVE_NAME_wav}{ARCHIVE_EXT}", target_dir, ARCHIVE_URL_wav)
+    # Conditionally extract data
+    _maybe_extract(target_dir, f"{ARCHIVE_ROOT_PATH}/{ARCHIVE_NAME_txt}", txt_archive_path)
+    _maybe_extract(target_dir, f"{ARCHIVE_ROOT_PATH}/{ARCHIVE_NAME_wav}", wav_archive_path)
+    # Produce CSV files
+    _maybe_convert_sets(target_dir, ARCHIVE_ROOT_PATH)
+
+
+def _maybe_extract(target_dir, extracted_data, archive_path):
+    # If target_dir/extracted_data does not exist, extract archive in target_dir
+    try:
+        extracted_path = os.path.join(target_dir, extracted_data)
+        if os.path.exists(extracted_path):
+            print('Found directory "%s" - not extracting it from archive.' % extracted_path)
+        else:
+            raise FileNotFoundError
+    except FileNotFoundError:
+        print('No directory "%s" - extracting archive...' % extracted_path)
+        tar = tarfile.open(archive_path)
+        tar.extractall(target_dir)
+        tar.close()
+
+def one_sample(sample):
+    """ Take a audio file, and optionally convert it to 16kHz mono channel WAV """
+    wav_filename = sample[0]
+    original_wav = wav_filename
+    formatted_wav = os.path.splitext(wav_filename)[0] + "_.wav"
+    _maybe_convert_wav(original_wav, formatted_wav)
+    wav_filename = formatted_wav
+    file_size = -1
+    frames = 0
+    if os.path.exists(wav_filename):
+        file_size = os.path.getsize(wav_filename)
+        frames = int(
+            subprocess.check_output(
+                ["soxi", "-s", wav_filename], stderr=subprocess.STDOUT
+            )
+        )
+    label = label_filter(sample[1])
+    counter = get_counter()
+    rows = []
+    if file_size == -1:
+        # Excluding samples that failed upon conversion
+        counter["failed"] += 1
+    elif label is None:
+        # Excluding samples that failed on label validation
+        counter["invalid_label"] += 1
+    elif int(frames / SAMPLE_RATE * 1000 / 15 / 2) < len(str(label)):
+        # Excluding samples that are too short to fit the transcript
+        counter["too_short"] += 1
+    elif frames / SAMPLE_RATE > MAX_SECS:
+        # Excluding very long samples to keep a reasonable batch-size
+        counter["too_long"] += 1
+    else:
+        # This one is good - keep it for the target CSV
+        rows.append((wav_filename, file_size, label))
```

Comment:
I would prefer to produce `rows.append((wav_filename.relative_to(os.path.join(target_dir, ARCHIVE_ROOT_PATH)), file_size, label))` to get relative path in the csv files.